server/src/dtos/album.dto.ts

```diff
@@ -109,13 +124,29 @@ export class AlbumResponseDto {
 
 export const mapAlbum = (entity: AlbumEntity, withAssets: boolean, auth?: AuthDto): AlbumResponseDto => {
   const sharedUsers: UserResponseDto[] = [];
-
-  if (entity.sharedUsers) {
-    for (const user of entity.sharedUsers) {
-      sharedUsers.push(mapUser(user));
+  const albumUsers: AlbumUserResponseDto[] = [];
+
+  if (entity.albumUsers) {
+    for (const albumUser of entity.albumUsers) {
+      const user = mapUser(albumUser.user);
+      sharedUsers.push(user);
+      albumUsers.push({
+        user,
+        role: albumUser.role,
+      });
     }
   }
 
+  albumUsers.sort((a, b) => {
+    if (a.role === AlbumUserRole.VIEWER && b.role === AlbumUserRole.EDITOR) {
+      return 1;
+    }
+    if (a.role === AlbumUserRole.EDITOR && b.role === AlbumUserRole.VIEWER) {
+      return -1;
+    }
+    return a.user.name.localeCompare(b.user.name);
+  });
+
```

Comment:
can you just use lodash orderBy? `_.orderBy(albumUsers, ['role', 'user.name'], ['DESC', 'ASC'])`